packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -87,19 +93,69 @@ export function SelectPanel({
   textInputProps,
   overlayProps,
   sx,
+  loading,
+  initialLoadingType = 'spinner',
   ...listProps
 }: SelectPanelProps): JSX.Element {
+  const inputRef = React.useRef<HTMLInputElement>(null)
   const titleId = useId()
   const subtitleId = useId()
+  const dataLoadedOnce = useRef(false)
```

Comment:
Could you tell me a little bit more about the reason of initialising `dataLoadedOnce` as a ref instead of a normal state variable?  Is it because to compare obj references?

Reply:
Hahaha that's probably because I don't actually understand the difference between the two since I don't write React code every day 😅 I can switch it to use `useState`, that's a good idea 👍